File: backend/app/schemas/booking.py
from enum import Enum
from typing import Optional, List
from uuid import UUID
from datetime import datetime
from pydantic import BaseModel, Field, ConfigDict


class BookingStatus(str, Enum):
    PENDING = "pending"
    ACCEPTED = "accepted"
    REJECTED = "rejected"
    CANCELLED = "cancelled"
    COMPLETED = "completed"


class BookingCreate(BaseModel):
    worker_id: UUID = Field(..., description="Target worker UUID")
    service_request_id: UUID = Field(..., description="Service request UUID to book")
    scheduled_time: Optional[datetime] = Field(None, description="Optional scheduled engagement timestamp")
    notes: Optional[str] = Field(None, max_length=1000, description="Optional customer booking notes")


class BookingStatusUpdate(BaseModel):
    status: BookingStatus = Field(..., description="Target status ('accepted' or 'rejected')")


class BookingResponse(BaseModel):
    booking_id: UUID
    customer_id: UUID
    worker_id: UUID
    service_request_id: Optional[UUID] = None
    customer_name: Optional[str] = None
    worker_name: Optional[str] = None
    worker_rating: Optional[float] = None
    description: Optional[str] = None
    category: Optional[str] = None
    urgency: Optional[str] = None
    scheduled_time: Optional[datetime] = None
    status: BookingStatus
    notes: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class BookingListResponse(BaseModel):
    total: int
    limit: int
    offset: int
    items: List[BookingResponse]

File: web/types/booking.ts
export type BookingStatus =
  | "pending"
  | "accepted"
  | "rejected"
  | "cancelled"
  | "completed";

export interface Booking {
  booking_id: string;
  customer_id: string;
  worker_id: string;
  service_request_id?: string | null;
  customer_name?: string | null;
  worker_name?: string | null;
  worker_rating?: number | null;
  description?: string | null;
  category?: string | null;
  urgency?: string | null;
  scheduled_time?: string | null;
  status: BookingStatus;
  notes?: string | null;
  created_at: string;
  updated_at: string;
}

export interface BookingListResponse {
  total: number;
  limit: number;
  offset: number;
  items: Booking[];
}

export interface BookingCreateInput {
  worker_id: string;
  service_request_id: string;
  scheduled_time?: string | null;
  notes?: string | null;
}

export interface BookingStatusUpdateInput {
  status: "accepted" | "rejected";
}
